source/Octostache/Templates/Functions/TextEncryptFunction.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+
+namespace Octostache.Templates.Functions
+{
+    internal class TextEncryptFunction
+    {
+        public static string RSAEncrypt(string data, string[] arguments)
+        {
+            if (arguments == null || arguments.Length == 0 || arguments.Length > 2)
+            {
+                return null;
+            }
+
+            try
+            {
+
+                var certificate = new X509Certificate2(Convert.FromBase64String(arguments[0]));
```

Comment:
The certificate is not disposed. Also we should think what key set to use and which store. `Ephemeral` seems to be a good choice in this, but I'm not sure of that.

Reply:
Looks like the Ephemeral key store is not natively availabe in .Net Standard 1.3 or .Net Framework 4.0. It seems to work if I use:`new X509Certificate2(Convert.FromBase64String(arguments[0]), (string)null, (X509KeyStorageFlags)32)` but I am not confident it would work on a machine that does not have a later version of .Net installed.